[check-webkit-style] Allow `[[likely|unlikely]]` after `else`

sosukesuzuki opened the pull request:
#### ac5c0c047f1b3759a3214a2c71efdedf18003bcb
<pre>
[check-webkit-style] Allow `[[likely|unlikely]]` after `else`
<a href="https://bugs.webkit.org/show_bug.cgi?id=294505">https://bugs.webkit.org/show_bug.cgi?id=294505</a>

Reviewed by Chris Dumez.

This patch changes `check-webkit-style` to `[[likely]]` / `[[unlikely]]`
after `else` token. Like...

```cpp
if (condition) {
    foo();
    bar();
} else [[likely]] {
    foo();
    bar();
}
````

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_spacing):
(check_braces):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):

Canonical link: <a href="https://commits.webkit.org/296238@main">https://commits.webkit.org/296238@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1af7b227eea1df251c017a48d79e6d89c8df6809

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107935 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27600 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18019 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113146 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58457 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109898 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28295 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36149 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81925 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22420 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97244 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62356 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/107367 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21832 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15377 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57899 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91771 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15443 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116273 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35007 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25762 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90958 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35383 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93522 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90752 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23114 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35656 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13409 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30762 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34905 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34648 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38008 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36309 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2de2d1dfd5e8ce69b09db4f27a17cc24911959a4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46762

sosukesuzuki commented on Tools/Scripts/webkitpy/style/checkers/cpp.py:
```diff
@@ -3150,7 +3150,8 @@ def check_braces(clean_lines, line_number, file_state, error):
         and not search(r'\belse if\b', line)
-        and not search(r'\belse\s*\\$', line)):
+        and not search(r'\belse\s*\\$', line)
+        and not search(r'\belse\s*\[\[(likely|unlikely)\]\]', line)):
```
`check-webkit-style` reports this line with an error: `ERROR: Tools/Scripts/webkitpy/style/checkers/cpp.py:3154:  visually indented line with same indent as next logical line  [pep8/E129] [5]`. However, since the original code and similar code elsewhere follow this style, I believe it should remain.

cdumez replied:
Please fix failure before landing:
```
ERROR: Tools/Scripts/webkitpy/style/checkers/cpp.py:3154:  visually indented line with same indent as next logical line  [pep8/E129] [5]
```

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1af7b227eea1df251c017a48d79e6d89c8df6809. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46762

sosukesuzuki committed:
[check-webkit-style] Allow `[[likely|unlikely]]` after `else`
https://bugs.webkit.org/show_bug.cgi?id=294505

Reviewed by Chris Dumez.

This patch changes `check-webkit-style` to `[[likely]]` / `[[unlikely]]`
after `else` token. Like...

```cpp
if (condition) {
    foo();
    bar();
} else [[likely]] {
    foo();
    bar();
}
````

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_spacing):
(check_braces):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):

Canonical link: https://commits.webkit.org/296238@main
